domain/ide/workspace/workspace.go

```diff
@@ -124,16 +124,20 @@ func (w *Workspace) ScanWorkspace(ctx context.Context) {
 	}
 }
 
-func (w *Workspace) AddAndRemoveFoldersAndTriggerScan(ctx context.Context, params lsp.DidChangeWorkspaceFoldersParams) {
+// ChangeWorkspaceFolders clears the "Removed" folders, adds the "New" folders,
+// and starts an automatic scan if auto-scans are enabled.
+func (w *Workspace) ChangeWorkspaceFolders(ctx context.Context, params lsp.DidChangeWorkspaceFoldersParams) {
```

Comment:
The old name was more explicit and easier to understand, IMHO.

Reply:
But now it's `AddAndRemoveFoldersAndTriggerScanIfAutoScanIsEnabled`

Reply:
It doesn't need to cover all conditions in the name ;). Right now, e.g. trust is not in there either.